src/main/java/com/strangeone101/holoitemsapi/enchantment/EnchantListener.java

```diff
@@ -261,23 +295,38 @@ private static Map<Enchantment, Integer> combineCustomEnchants(Map<Enchantment,
             ));
     }
 
+    /**
+     * Get the total enchantment cost from the map of enchantments.
+     * @param enchantments Enchantments to calculate level cost
+     * @param initial Initial value to add from
+     * @return Level cost needed to add enchantments to an item
+     */
+    private static int getCustomEnchantCost(Map<Enchantment, Integer> enchantments, int initial) {
+        return enchantments.entrySet().stream()
+            .reduce(initial,
+                (num, entry) -> num + ((CustomEnchantment) entry.getKey()).getLevelMultiplier() * entry.getValue(),
+                Integer::sum);
+    }
+
     private static boolean hasNoConflictEnchants(Map<Enchantment, Integer> enchants, Enchantment filter) {
         return enchants.keySet().stream()
             .noneMatch(filter::conflictsWith);
     }
 
     private static boolean hasNoCustomEnchants(@Nullable ItemStack itemStack) {
-        return itemStack == null || !itemStack.hasItemMeta()
-            || itemStack.getEnchantments().keySet().stream()
-                .noneMatch(CustomEnchantment.class::isInstance);
+        if (itemStack == null || !itemStack.hasItemMeta())
+            return true;
+
+        if (itemStack.getItemMeta() instanceof EnchantmentStorageMeta enchantmentMeta) {
+            return enchantmentMeta.getStoredEnchants().entrySet().stream()
+                .noneMatch(entry -> entry.getKey() instanceof CustomEnchantment);
+        }
+
+        return itemStack.getEnchantments().entrySet().stream()
+            .noneMatch(entry -> entry.getKey() instanceof CustomEnchantment);
     }
 
-    /**
-     * method to cap level cost. Any level cost above 39 in anvils results in "Too Expensive" to be shown.
-     * @param level The current level cost
-     * @return A level lower than 40
-     */
-    private static int capLevelCost(int level) {
-        return Integer.min(level, 39);
+    private static boolean hasEnoughLevels(Player player, int repairCost) {
+        return !(repairCost >= MAX_REPAIR_COST || player.getLevel() < repairCost);
```

Comment:
How about we invert this condition?
```suggestion
        return repairCost <= player.getLevel() && repairCost < MAX_REPAIR_COST;
```
Rather, is it even necessary to check against MAX_REPAIR_COST? Who is going to get more than 32,767 levels?

Reply:
Honestly, thinking back i realized that no combination of custom enchants **should** even result in a level cost higher than that.